public/javascripts/SVLabel/build/SVLabel.js

```diff
@@ -11735,39 +11747,207 @@ function ModalMissionComplete ($, d3, L) {
+            var  geojsonFeature = missionTasks[i].getFeature();
+            var layer = L.geoJson(geojsonFeature).addTo(map);
+            layer.setStyle(missionTaskLayerStyle);
+        }
+    }
+    /**
+     * This method takes tasks that has been completed in the current mission and *all* the tasks completed in the
+     * current neighborhood so far.
+     * WARNING: `completedTasks` include tasks completed in the current mission too.
+     * WARNING2: The current tasks are not included in neither of `missionTasks` and `completedTasks`
+     *
+     * @param missionTasks
+     * @param completedTasks
+     * @private
+     */
     function _updateNeighborhoodStreetSegmentVisualization(missionTasks, completedTasks) {
         // var completedTasks = svl.taskContainer.getCompletedTasks(regionId);
         // var missionTasks = mission.getRoute();
 
         if (completedTasks && missionTasks) {
             // Add layers http://leafletjs.com/reference.html#map-addlayer
-            var i, len, geojsonFeature, layer,
-                completedTaskLayerStyle = { color: "rgb(128, 128, 128)", opacity: 1, weight: 3 },
-                missionTaskLayerStyle = { color: "rgb(49,130,189)", opacity: 1, weight: 3 };
+            var i, len, geojsonFeature, featureCollection, layer,
+                completedTaskLayerStyle = { color: "rgb(49,130,189)", opacity: 1, weight: 3 };
 
-            // Add the completed task layer
+            // remove after animation, otherwise segments remain green from previous tasks
+            d3.select(map.getPanes().overlayPane)
+                .selectAll("svg")
+                .selectAll(".lineConnect")
+                .remove();
+
+            var newStreets = streetEdgeIds(missionTasks);
             len = completedTasks.length;
+            // Add the completed task layer
             for (i = 0; i < len; i++) {
-                geojsonFeature = completedTasks[i].getFeature();
-                layer = L.geoJson(geojsonFeature).addTo(map);
-                layer.setStyle(completedTaskLayerStyle);
+                var streetEdgeId = completedTasks[i].getStreetEdgeId();
+                // do not render streets already in blue
+                if(newStreets.indexOf(streetEdgeId) == -1){
+                    geojsonFeature = completedTasks[i].getFeature();
+                    layer = L.geoJson(geojsonFeature).addTo(map);
+                    layer.setStyle(completedTaskLayerStyle);
+                }
             }
-
+            
             // Add the current mission layer
+            
             len = missionTasks.length;
-            for (i = 0; i < len; i++) {
-                geojsonFeature = missionTasks[i].getFeature();
-                layer = L.geoJson(geojsonFeature).addTo(map);
-                layer.setStyle(missionTaskLayerStyle);
-            }
+            // disable zooming while animating, gets re-enabled in animation function
+            map.touchZoom.disable();
```

Comment:
Why do we have to disable? I thought you fixed the animation so that zooming does not affect animation.